qualtran/bloqs/factoring/rsa/rsa_mod_exp.py

```diff
@@ -85,33 +87,39 @@ def signature(self) -> 'Signature':
         )
 
     @classmethod
-    def make_for_shor(cls, big_n: int, g: Optional[int] = None):
+    def make_for_shor(cls, big_n: 'SymbolicInt', g: Optional['SymbolicInt'] = None):
         """Factory method that sets up the modular exponentiation for a factoring run.
 
         Args:
             big_n: The large composite number N. Used to set `mod`. Its bitsize is used
                 to set `x_bitsize` and `exp_bitsize`.
             g: Optional base of the exponentiation. If `None`, we pick a random base.
         """
-        if isinstance(big_n, sympy.Expr):
+        if is_symbolic(big_n):
             little_n = sympy.ceiling(sympy.log(big_n, 2))
         else:
             little_n = int(math.ceil(math.log2(big_n)))
         if g is None:
-            g = random.randint(2, big_n)
+            if is_symbolic(big_n):
+                g = sympy.symbols('g')
+            else:
+                while True:
+                    g = random.randint(2, int(big_n))
```

Comment:
thanks for fixing this, but I'm going to be very picky:

please add an optional `rs: np.RandomState = None` argument to this method that can make this deterministic. If it's not provided, make a new `np.random.RandomState()`. Use a seeded, deterministic random state in all tests.